queue/remote_base.go

```diff
@@ -13,8 +13,13 @@ import (
 	"github.com/pkg/errors"
 )
 
+// remoteQueue is an interface to an amboy.RetryableQueue that uses a
+// remoteQueueDriver to interact with the persistence layer for the queue.
 type remoteQueue interface {
 	amboy.RetryableQueue
+	// SetDriver sets the driver to connect to the persistence layer. The driver
+	// must be set before the queue can start. Once the queue has started, the
+	// driver cannot be modified.
 	SetDriver(remoteQueueDriver) error
 	Driver() remoteQueueDriver
```

Comment:
Should we add documentation here since it is exported?

Reply:
It's not exported because `remoteQueue` is not exported (I have no idea why), but they could be unexported if we wanted. Even though we don't need them, I added docs for completeness and so that others can understand the interface just from the doc comments.